### PR TITLE
Covid-19 - Metrics

### PR DESCRIPTION
**Pull request by elisha2kyakpo1**
In this branch, I implemented the following tasks:

1. Created a repo and configured linters.
2. Created redux and react components.
3. Installed project dependencies.
4. Chose an API for the corona data.
5. Created the redux store and made all the data dispatching.
6. Tested pure functions.
7. Added styles for the UI.
8. Deployed the app to the server.

## Loom video
[Project presentation](https://loom.com/share/d620617bd1fe40e1b9b2c45f74c79801)

### CODE REVIEW

File: src/components/Metrics.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+              <Item confirmed={formatNumber(confirmed)} name={name} />
+            </Link>
+          </div>
+        </div>
+      ))}
+    </div>
+  </div>
+);
+
+const Metrics = () => {
```
Comment (Georjane) [inline src/components/Metrics.js]: - [ ] Your project looks very nice! As per the project requirement, you should add a Filter stateless component to filter data retrieved from the API. For example, you can add a functionality to filter the data by country, so a user can search for a country

**Review by Georjane — CHANGES_REQUESTED** [review]
### Hello, Team <STUDENTS>, ✋🏽 

Good job so far! You have a very professional Readme file, your PR is professional and you have a good mastery of git-flow. Your linters are all passing, I applaud that too.
However, we have a few more changes to make in order to meet all requirements for this project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding! 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 🤝 

_Code Reviewer
**[Jane](https://witahgeorjane.tech/)**_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Metrics.js (hunk 1 above)
Comment (elisha2kyakpo1) [reply src/components/Metrics.js]: Hello @Georjane, thanks for the review, please try to explain to me what you mean. Are you saying that you did not navigate to the details page or what? The data is being filtered that's the reason why I created two fetch functions to filter by continent or by country

Comment (Georjane) [reply src/components/Metrics.js]: Hi again @elisha2kyakpo1, yes you are right. You did filter the fetch data before displaying it on the home page. 
I was rather looking for a stateless filter component which is what was required but I think you filtered the data using a stateful component.

- [ ] Can you please separate the logic so your App uses a Filter stateless component to filter data retrieved from the API.

Comment (elisha2kyakpo1) [reply src/components/Metrics.js]: Hi, @Georjane, data is made available through props

**Review by oloomoses — CHANGES_REQUESTED** [review]
Hi @elisha2kyakpo1, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
- [ ] Kindly create a filter component to allow a user to filter data with some parameters as per the project requirements.
Screenshot from 2021-09-09 20-23-55

- [ ] The filter should be a stateless component
Screenshot from 2021-09-09 20-23-02


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Slack: Oloo Moses_
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by elisha2kyakpo1** [comment]
Hello @oloomoses and @Georjane, thank you so much for the review. I would like you to take a look at my ItemDetails.js component, I'm filtering by country, see from this image. In this case, I'm getting a specific country name to be dispatched to the store, the fetchCountry(name), the name is a parameter passed to the fetch API URL, I created a separate file called Stateless.js so that you can see how I'm fetching data by filtering.

stateleess

API-data


This also applies to the continent. I'm not going to do anything apart from what I have done. Go through it properly and give me feedback before I contact student success.

**Comment by elisha2kyakpo1** [comment]
This is a mobile app, a user clicks the country image, that's how they filter data based on country. there is no input field. when you click on the country you want to filter, the country name will be passed to the fetchCountry function. If there is something I'm missing, I can give you a zoom link you show me.

**Comment by oloomoses** [comment]
> This is a mobile app, a user clicks the country image, that's how they filter data based on country. there is no input field. when you click on the country you want to filter, the country name will be passed to the fetchCountry function. If there is something I'm missing, I can give you a zoom link you show me.

Hi @elisha2kyakpo1, thanks for reaching out. The project requirement is very clear that when a visitor lands on the home page, they should be able to filter the items by selecting or entering some values. Kindly check this [project](https://www.akasharojee.codes/stock-screener/) to get an idea.

**Comment by elisha2kyakpo1** [comment]
Hi, @oloomoses, here is a zoom link, https://microverse.zoom.us/j/92429453054?pwd=TXg1djZQblRoemJubDQ1dHpUZ3VKdz09

Join me just less than 5minutes

**Comment by elisha2kyakpo1** [comment]
@oloomoses, I'm already on the call waiting

**Review by mricanho — CHANGES_REQUESTED** [review]
Hi @elisha2kyakpo1, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️

- [ ] The requirement the other reviewers are talking about is category filter you can filter it by name or by whatever variable you want to use, but it's a hard requirement, you need to do this in order to approve the capstone, this new feature must be in the home page, not in the detail page; all the other aspects of your project are ok, you don't need to modify anything else. Please, add it.

@oloomoses  already gives you a project example in the comments.

Requirement:

imagen


_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikenath223 — APPROVED** [review]
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**